proxy/envoy/resources.go

```diff
@@ -175,12 +186,17 @@ type HTTPRoute struct {
 	faults []*HTTPFilter
 }
 
+// CatchAll returns true if the route matches all requests
+func (route *HTTPRoute) CatchAll() bool {
+	return len(route.Headers) == 0 && route.Path == "" && route.Prefix == "/"
+}
```

Comment:
The function assumes the route is valid (one of path or prefix is set). It looks like we set path and prefix in every route soon after instantiation. It's probably a good idea to move it to a factory method so that no invalid routes are created.